### DISCUSSION TITLE
Generated fields and useFieldArray

### DISCUSSION THREAD

**Discussion by douglashoang1** [react-hook-form/react-hook-form#3622] [Q&A] [answered]
Hi I'm trying to see what the best way to generate dynamic fields based on the values within a field array.

I have a basic codepen up here https://codesandbox.io/s/react-hook-form-usefieldarray-forked-0yg85?file=/src/index.js.

Would it be better for me to just watch the individual values within the array instead of the whole array?

Thanks :)

**Comment by bluebill1049** [comment, marked as answer]
it's better to use `useWatch` if u have perf issue.

**Reply by douglashoang1** [reply]
I'm currently using `useWatch` but I'm running into an infinite loop when I try to generate the dynamic fields.

Oops should have added that to my initial question.

**Reply by bluebill1049** [reply]
can you share a codesandbox?

**Reply by douglashoang1** [reply]
` 
  useFieldArray({
    control,
    name: "test1",
    defaultValue: [{ name: "a" }, { name: "b" }]
  });

  const { fields, append, remove } = useFieldArray({
    control,
    name: "test"
  });

  const watchTest1 = useWatch({
    name: "test1",
    control,
    defaultValue: ["a", "b"]
  });

  useEffect(() => {
    console.log("infinite loop");
    // I want to generate fields based on the array test1
    reset({
      ...getValues(),
      test: watchTest1.map((t) => ({
        firstName: t.name,
        lastName: `${t.name}-hi`
      }))
    });
  }, [
    watchTest1,
    getValues,
    reset
  ]);`

I'm running into an infinite loop with this code above. I'm trying to watch the array in `test1` and based on the values within that array I want to generate dynamic fields with `test`

**Reply by douglashoang1** [reply]
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-0yg85?file=/src/index.js

**Reply by bluebill1049** [reply]
`useWatch` result is not memorized inside hook form.

**Reply by douglashoang1** [reply]
Does `watch` memorize the result or are there any plans on memorizing the result for the `useWatch` hook?

**Reply by douglashoang1** [reply]
I think I found a solution that works for me that doesn't require the result being memorized. I'm using `useWatch` to watch the individual values like `test1[idx].firstName` and `test1[idx].firstName` instead of watching the entire fieldArray.

Thanks for the help :+1:

**Reply by bluebill1049** [reply]
oh cool, awesome.
